src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -1092,6 +1092,22 @@ pub extern "C" fn write_log(msg: *const raw::c_char, level: u64, target: *const
     .expect("Error logging message");
 }
 
+#[no_mangle]
+pub extern "C" fn write_stdout(session_ptr: *mut Session, msg: *const raw::c_char) {
+  with_session(session_ptr, |session| {
+    let message_str = unsafe { CStr::from_ptr(msg).to_string_lossy() };
```

Comment:
Could you add a TODO here to port this over to the `EngineDisplay`? Just in case.